File: src/androidTest/java/com/nextcloud/android/lib/resources/directediting/DirectEditingObtainRemoteOperationIT.java
/*
 *
 * Nextcloud Android client application
 *
 * @author Tobias Kaminsky
 * Copyright (C) 2019 Tobias Kaminsky
 * Copyright (C) 2019 Nextcloud GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 */

package com.nextcloud.android.lib.resources.directediting;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import com.owncloud.android.AbstractIT;
import com.owncloud.android.lib.common.Creator;
import com.owncloud.android.lib.common.DirectEditing;
import com.owncloud.android.lib.common.Editor;
import com.owncloud.android.lib.common.operations.RemoteOperationResult;

import org.junit.Test;

public class DirectEditingObtainRemoteOperationIT extends AbstractIT {

    @Test
    public void testGetAll() {
        RemoteOperationResult<DirectEditing> result = new DirectEditingObtainRemoteOperation().execute(client);
        assertTrue(result.isSuccess());

        DirectEditing directEditing = result.getResultData();

        assertTrue(directEditing.getEditors().containsKey("text"));

        Editor textEditor = directEditing.getEditors().get("text");
        assertNotNull(textEditor);

        assertEquals("Nextcloud Text", textEditor.getName());

        assertTrue(textEditor.getMimetypes().contains("text/markdown"));
        assertTrue(textEditor.getMimetypes().contains("text/plain"));
        assertEquals(0, textEditor.getOptionalMimetypes().size());

        Creator creator = directEditing.getCreators().get("textdocument");
        assertNotNull(creator);
        assertFalse(creator.templates);
    }
}

File: library/src/androidTest/java/com/nextcloud/android/lib/resources/directediting/DirectEditingObtainRemoteOperationIT.kt
/*
 * Nextcloud Android Library
 *
 * SPDX-FileCopyrightText: 2022-2024 Nextcloud GmbH and Nextcloud contributors
 * SPDX-FileCopyrightText: 2022 Tobias Kaminsky <tobias@kaminsky.me>
 * SPDX-License-Identifier: MIT
 */
package com.nextcloud.android.lib.resources.directediting

import com.owncloud.android.AbstractIT
import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertFalse
import junit.framework.Assert.assertNotNull
import junit.framework.Assert.assertTrue
import org.junit.Test

class DirectEditingObtainRemoteOperationIT : AbstractIT() {
    @Test
    fun testGetAll() {
        val result = DirectEditingObtainRemoteOperation().run(nextcloudClient)
        assertTrue(result.isSuccess)

        val (editors, creators) = result.resultData
        assertTrue(editors.containsKey("text"))

        val textEditor = editors["text"]
        assertNotNull(textEditor)
        assertEquals("Nextcloud Text", textEditor!!.name)
        assertTrue(textEditor.mimetypes.contains("text/markdown"))
        assertTrue(textEditor.mimetypes.contains("text/plain"))
        assertEquals(0, textEditor.optionalMimetypes.size.toLong())

        val creator = creators["textdocument"]
        assertNotNull(creator)
        assertFalse(creator!!.templates)
    }
}
